wasm/src/language_service.rs

```diff
@@ -33,17 +33,25 @@ impl LanguageService {
         LanguageService(inner)
     }
 
-    pub fn update_document(&mut self, uri: &str, version: u32, text: &str, is_exe: bool) {
-        self.0.update_document(
-            uri,
-            version,
-            text,
-            if is_exe {
-                qsc::PackageType::Exe
-            } else {
-                qsc::PackageType::Lib
-            },
-        );
+    pub fn update_configuration(&mut self, config: IWorkspaceConfiguration) {
+        let config: WorkspaceConfiguration = config.into();
+        self.0
+            .update_configuration(&qsls::protocol::WorkspaceConfigurationUpdate {
+                target_profile: config.targetProfile.map(|s| match s.as_str() {
+                    "base" => qsc::TargetProfile::Base,
+                    "full" => qsc::TargetProfile::Full,
+                    _ => panic!("invalid target profile"),
+                }),
+                package_type: config.packageType.map(|s| match s.as_str() {
+                    "lib" => qsc::PackageType::Lib,
+                    "exe" => qsc::PackageType::Exe,
+                    _ => panic!("invalid package type"),
```

Comment:
Being the string is caller provided data, is a panic on an unexpected string the right behavior here? Or would returning an Err be more appropriate (which wasm turns into an exception on the JavaScript side).

Reply:
Hmmm good question. The TypeScript type of `IWorkspaceConfiguration.packageType` is defined as `"lib" | "exe"`, so the correct value will be "enforced" by the .d.ts. And the wasm library is only ever meant to be used from our own npm package. In the VS Code extension, I sanitize the user-provided setting before passing it into this method. So I feel like the developer would be _really_ trying to shoot themselves in the foot if a wrong value slips in, and I feel that the panic suits the fail-fast approach I tend to prefer.

I like it this way now, but I can be talked out of it if it's important.